src/Chapter3.hs

```diff
@@ -1006,6 +1143,187 @@ Implement data types and typeclasses, describing such a battle between two
 contestants, and write a function that decides the outcome of a fight!
 -}
 
+data Action = T9Attack | Potion | Spell | Runaway  deriving (Show, Eq)
```

Comment:
It would be recommended to create separate data types for `Knight` action and `Monster` action so it won't be possible to mix actions of different types :slightly_smiling_face: